core/src/test/scala/org/apache/spark/storage/ShuffleBlockFetcherIteratorSuite.scala

```diff
@@ -364,6 +368,72 @@ class ShuffleBlockFetcherIteratorSuite extends SparkFunSuite with PrivateMethodT
     assert(blockManager.hostLocalDirManager.get.getCachedHostLocalDirs.size === 1)
   }
 
+  test("BlockStoreClient getHostLocalDirs RPC supports IOException retry") {
+    val mockClientFactory = mock(classOf[TransportClientFactory])
+    val mockTransportClient = mock(classOf[TransportClient])
+    val execToDirs = Map("exec-1" ->
+      Array("loc2.1", "loc2.2"))
+
+    var sendRpcThrowIOExceptionIdx = 0
+    var sendRpcThrowIOExceptionMaxCnt = 2
```

Comment:
The initial value is a bit confusing - it is always reset below.

```suggestion
    var sendRpcThrowIOExceptionMaxCnt = 0
```